stm32/libcanard/canard_stm32.c

```diff
@@ -246,33 +251,33 @@ int16_t canardSTM32Init(const CanardSTM32CANTimings* const timings,
      * MCU within the STM32 family.
      */
     {
-        uint32_t fmr = CANARD_STM32_CAN1->FMR & 0xFFFFC0F1U;
+        uint32_t fmr = FILTER_CONFIG_BXCAN->FMR & 0xFFFFC0F1U;
         fmr |= CANARD_STM32_NUM_ACCEPTANCE_FILTERS << 8U;                // CAN2 start bank = 14 (if CAN2 is present)
-        CANARD_STM32_CAN1->FMR = fmr | CANARD_STM32_CAN_FMR_FINIT;
+        FILTER_CONFIG_BXCAN->FMR = fmr | CANARD_STM32_CAN_FMR_FINIT;
     }
 
-    CANARD_ASSERT(((CANARD_STM32_CAN1->FMR >> 8U) & 0x3FU) == CANARD_STM32_NUM_ACCEPTANCE_FILTERS);
+    CANARD_ASSERT(((FILTER_CONFIG_BXCAN->FMR >> 8U) & 0x3FU) == CANARD_STM32_NUM_ACCEPTANCE_FILTERS);
 
-    CANARD_STM32_CAN1->FM1R = 0;                                        // Indentifier Mask mode
-    CANARD_STM32_CAN1->FS1R = 0x0FFFFFFF;                               // All 32-bit
+    FILTER_CONFIG_BXCAN->FM1R = 0;                                       // Identifier Mask mode
+    FILTER_CONFIG_BXCAN->FS1R = 0x0FFFFFFF;                              // All 32-bit
 
     // Filters are alternating between FIFO0 and FIFO1 in order to equalize the load.
     // This will cause occasional priority inversion and frame reordering on reception,
     // but that is acceptable for UAVCAN, and a majority of other protocols will tolerate
     // this too, since there will be no reordering within the same CAN ID.
-    CANARD_STM32_CAN1->FFA1R = 0x0AAAAAAA;
+    FILTER_CONFIG_BXCAN->FFA1R = 0x0AAAAAAA;
 
 #if CANARD_STM32_USE_CAN2
     CANARD_STM32_CAN1->FilterRegister[CANARD_STM32_NUM_ACCEPTANCE_FILTERS].FR1 = 0;
     CANARD_STM32_CAN1->FilterRegister[CANARD_STM32_NUM_ACCEPTANCE_FILTERS].FR2 = 0;
     CANARD_STM32_CAN1->FA1R = (1 << CANARD_STM32_NUM_ACCEPTANCE_FILTERS);  // One filter enabled
 #else
-    CANARD_STM32_CAN1->FilterRegister[0].FR1 = 0;
-    CANARD_STM32_CAN1->FilterRegister[0].FR2 = 0;
-    CANARD_STM32_CAN1->FA1R = 1;                                        // One filter enabled
+    BXCAN->FilterRegister[0].FR1 = 0;
+    BXCAN->FilterRegister[0].FR2 = 0;
+    BXCAN->FA1R = 1;                                        // One filter enabled
```

Comment:
I suggest we use `FILTER_CONFIG_BXCAN` here for clarity. The logic would not change but the intent should be more obvious.